src/plugins/telemetry_collection_manager/server/cache/cache_manager.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import moment from 'moment';
+export interface CachedObject<T = unknown> {
+  data: T;
+  cacheTimestamp: string;
+}
+
+export interface CacheManagerConfig {
+  cacheDurationMs: number;
+}
+
+export class CacheManager {
```

Comment:
Can we use `lru-cache` already used in the repo? we can drastically reduce the number of tests and docs 😄

Reply:
++ to avoid reinventing the wheel if possible.